### PR TITLE
record job inspect in task result and reveal events

### PR DESCRIPTION
**Pull request by chengjoey**
#### What type of this PR

Add one of the following kinds:
/kind feature

#### What this PR does / why we need it:
record job inspect in task result
reveal job events

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  record job inspect in task result            |
| 🇨🇳 中文    |   记录job的inspect           |

### CODE REVIEW

**Comment by chengjoey** [comment]
image

**Review by sfwn — CHANGES_REQUESTED** [review]

File: apistructs/k8s.go [hunk 1]
```diff
@@ -27,4 +27,5 @@ const (
 type TaskInspect struct {
 	Object runtime.Object    `json:"object"`
```
Comment (sfwn) [inline apistructs/k8s.go:28]: It's not common for all pipeline jobs. It's just for k8s jobs.

File: modules/pipeline/pipengine/actionexecutor/types/define.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ type ActionExecutor interface {
 
 	// Status 只做简单重试
 	Status(ctx context.Context, action *spec.PipelineTask) (apistructs.PipelineStatusDesc, error)
-	Inspect(ctx context.Context, action *spec.PipelineTask) (interface{}, error)
+	Inspect(ctx context.Context, action *spec.PipelineTask) (apistructs.TaskInspect, error)
```
Comment (sfwn) [inline modules/pipeline/pipengine/actionexecutor/types/define.go:41]: adjust struct.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: apistructs/k8s.go (hunk 1 above)
Comment (chengjoey) [reply apistructs/k8s.go:28]: object filed may change to interface,keep desc filed string

**Review by sfwn — APPROVED** [review]

**Pull request merged**